src/v/pandaproxy/schema_registry/storage.h

```diff
@@ -675,6 +688,20 @@ struct consume_to_store {
         vassert(res.version == val.version, "Schema_version mismatch");
     }
 
+    void apply(config_key key, config_value val) {
+        vassert(key.magic == 0, "Config key magic is unknown");
```

Comment:
we use `result<>` a lot in this series but we switched to asserts here for a very specific error which is hard to test for.

Reply:
I've futurized it, and returned errors that way.